src/test/libcephfs/test.cc

```diff
@@ -109,6 +109,62 @@ TEST(LibCephFS, OpenReadWrite) {
   ceph_shutdown(cmount);
 }
 
+TEST(LibCephFS, RstatsPropagate) {
```

Comment:
It would be better to have this as a series of unit tests in a new test file like `qa/tasks/cephfs/test_rstats.py`. Each test should be small and focused. We willl want to evaluate the correctness of the propagation for a multitude of directory structures.